### PR TITLE
Add storage_volume_snapshots

### PR DESCRIPTION
**Pull request by grubernaut**
Also run `gofmt` on `ip_addresss_associations_test.go`

```
$ make testacc TEST=./compute TESTARGS="-run=TestAccStorageVolumeSnapshot_Lifecycle"
==> Checking that code complies with gofmt requirements...
ORACLE_ACC=1 go test -v ./compute -run=TestAccStorageVolumeSnapshot_Lifecycle -timeout 120m
=== RUN   TestAccStorageVolumeSnapshot_Lifecycle
--- PASS: TestAccStorageVolumeSnapshot_Lifecycle (12.04s)
PASS
ok      github.com/hashicorp/go-oracle-terraform/compute        12.065s
```

### CODE REVIEW

File: compute/storage_volume_snapshots.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+	// Optional, will be generated if not specified
+	Name string `json:"name,omitempty"`
+
+	// Whether or not the parent volume is bootable
+	// Optional
+	ParentVolumeBootable string `json:"parent_volume_bootable,omitempty"`
+
+	// Whether collocated or remote
+	// Optional, will be remote if unspecified
+	Property string `json:"property,omitempty"`
```
Comment (tombuildsstuff) [inline compute/storage_volume_snapshots.go]: could this become a Constant?

File: compute/storage_volume_snapshots.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+	// Optional
+	Description string `json:"description,omitempty"`
+
+	// Name of the snapshot
+	// Optional, will be generated if not specified
+	Name string `json:"name,omitempty"`
+
+	// Whether or not the parent volume is bootable
+	// Optional
+	ParentVolumeBootable string `json:"parent_volume_bootable,omitempty"`
```
Comment (tombuildsstuff) [inline compute/storage_volume_snapshots.go]: is this a Boolean value?

File: compute/storage_volume_snapshots.go [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+	// Description of the snapshot
+	Description string `json:"description"`
+
+	// The name of the machine image that's used in the boot volume from which this snapshot is taken
+	MachineImageName string `json:"machineimage_name"`
+
+	// Name of the snapshot
+	Name string `json:"name"`
+
+	// String indicating whether the parent volume is bootable or not
```
Comment (tombuildsstuff) [inline compute/storage_volume_snapshots.go]: Is this a Boolean value?

File: compute/storage_volume_snapshots.go [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+	ParentVolumeBootable string `json:"parent_volume_bootable"`
+
+	// Platform the snapshot is compatible with
+	Platform string `json:"platform"`
+
+	// String determining whether the snapshot is remote or collocated
+	Property string `json:"property"`
```
Comment (tombuildsstuff) [inline compute/storage_volume_snapshots.go]: As below: shouldn't this be a Constant of some sort?

File: compute/storage_volume_snapshots.go (hunk 2 above)
Comment (grubernaut) [inline compute/storage_volume_snapshots.go]: It's unfortunately a `"string"` of `"true"` or `"false"`

File: compute/storage_volume_snapshots.go (hunk 3 above)
Comment (grubernaut) [inline compute/storage_volume_snapshots.go]: No, it's a string.

**Pull request merged**